upf/solvers/tarski_grounder.py

```diff
@@ -0,0 +1,102 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""This module implements the grounder that uses tarski."""
+
+
+
+from functools import partial
+from typing import Callable, Tuple, Dict, List
+import tarski # type: ignore
+import upf
+import upf.interop
+from upf.interop.from_tarski import convert_tarski_formula
+from upf.model.problem_kind import full_classical_kind, full_numeric_kind
+from upf.model import Action, FNode
+from upf.solvers.grounder import lift_plan
+from upf.solvers.solver import Solver
+from upf.transformers import Grounder
+
+
+from tarski.grounding import LPGroundingStrategy, NaiveGroundingStrategy # type: ignore
+
+
+class TarskiGrounder(Solver):
+    """Implements the gounder that uses tarski."""
+    def __init__(self, **kwargs):
+        if len(kwargs) > 0:
+            raise
+
+    @staticmethod
+    def name() -> str:
+        return 'tarski_grounder'
+
+    @staticmethod
+    def is_grounder() -> bool:
+        return True
+
+    @staticmethod
+    def supports(problem_kind: 'upf.model.ProblemKind') -> bool:
+        supported_kind = upf.model.ProblemKind()
+        supported_kind.set_typing('FLAT_TYPING') # type: ignore
+        supported_kind.set_conditions_kind('NEGATIVE_CONDITIONS') # type: ignore
+        supported_kind.set_conditions_kind('DISJUNCTIVE_CONDITIONS') # type: ignore
+        supported_kind.set_conditions_kind('EQUALITY') # type: ignore
+        supported_kind.set_conditions_kind('EXISTENTIAL_CONDITIONS') # type: ignore
+        supported_kind.set_conditions_kind('UNIVERSAL_CONDITIONS') # type: ignore
+        supported_kind.set_effects_kind('CONDITIONAL_EFFECTS') # type: ignore
+        supported_kind.set_fluents_type('OBJECT_FLUENTS') # type: ignore
+        supported_kind.set_numbers('DISCRETE_NUMBERS') # type: ignore
+        supported_kind.set_fluents_type('NUMERIC_FLUENTS') # type: ignore
+        supported_kind.set_effects_kind('INCREASE_EFFECTS') # type: ignore
+        supported_kind.set_effects_kind('DECREASE_EFFECTS') # type: ignore
+        return problem_kind <= supported_kind
+
+    def ground(self, problem: 'upf.model.Problem') -> Tuple['upf.model.Problem', Callable[[upf.plan.Plan], upf.plan.Plan]]:
+        tarski_problem = upf.interop.convert_problem_to_tarski(problem)
+        actions = None
+        state_variables = None
+        #TODO: understand when LP works and when not and use the problem_kind instead of try-catch
+        try:
+            lpgs = LPGroundingStrategy(tarski_problem)
+            actions = lpgs.ground_actions()
```

Comment:
This command can throw a `ReachabilityLPUnsolvable` exception. We should handle it explicitly.